### ISSUE TITLE
error validating extension "iscsi-tools"

### ISSUE THREAD

**Issue by docbobo** [siderolabs/extensions#509]
I was just trying to build an 1.8.2 RK3588 with the latest iscsi-tools version (v0.1.6), but this fails with the following error:

```
◳ error validating extension "iscsi-tools": path "/usr/lib/udev/rules.d/50-iscsi-firmware-login.rules" is not allowed in extensions
Error: error validating extension "iscsi-tools": path "/usr/lib/udev/rules.d/50-iscsi-firmware-login.rules" is not allowed in extensions
```

Using v0.1.5 instead works without any issues.

**Comment by frezbo** [comment]
You'd have to use the image by reference as mentioned in the docs: https://github.com/siderolabs/extensions?tab=readme-ov-file#installing-extensions

**Comment by MatthewStockdale** [comment]
hey @docbobo, I had this same error, and it seems to be because my build script is looking for the arm64 version of the extension against the tagged release instead of getting the latest version which you would get by following the documentation.

```
❯ bash create-rk1-installer-image.sh
Determining Talos version...
Talos client version v1.8.3 found. We will use that version for the Talos nodes, too.
Logging Docker into GitHub with user matthewstockdale...
Login Succeeded
Creating Talos image ghcr.io/matthewstockdale/talos-rk1-installer:v1.8.3 with provided extensions...
profile ready:
arch: arm64
platform: metal
secureboot: false
version: v1.8.3
input:
  kernel:
    path: /usr/install/arm64/vmlinuz
  initramfs:
    path: /usr/install/arm64/initramfs.xz
  baseInstaller:
    imageRef: ghcr.io/nberlee/installer:v1.8.3-rk3588
  overlayInstaller:
    imageRef: ghcr.io/nberlee/sbc-turingrk1:v1.8.3
  systemExtensions:
    - imageRef: ghcr.io/nberlee/rk3588:v1.8.3@sha256:61311adeff52826e659c1a80ad49940241bb08f3aaece80157834cf7f75cb789
    - imageRef: ghcr.io/siderolabs/util-linux-tools:2.40.2@sha256:380740d159c024efa0f80136e7fdd18b36afdac77bad4f452aa83099ba115397
    - imageRef: ghcr.io/siderolabs/iscsi-tools:v0.1.6@sha256:e35f8291bfa88b7b334b5956d8d5048482f1dafbca0f7565102e59188c534716
    - imageRef: ghcr.io/siderolabs/nut-client:2.8.2@sha256:547104892638321a88e5332c7b357a9a986b5f142afd247b201b4c726a97ac97
overlay:
  name: turingrk1
  image:
    imageRef: ghcr.io/nberlee/sbc-turingrk1:v1.8.3
output:
  kind: installer
  outFormat: raw
◲ error validating extension "iscsi-tools": path "/usr/lib/udev/rules.d/50-iscsi-firmware-login.rules" is not allowed in extensions
Error: error validating extension "iscsi-tools": path "/usr/lib/udev/rules.d/50-iscsi-firmware-login.rules" is not allowed in extensions
*** Installer expected to be created by the Imager (previous step), but it is not! Exiting. ***
```

Like you, I had previously been able to use my script successfully with iscsi-tools v0.1.5.

I had to hardcode my script to use [this untagged version](https://github.com/siderolabs/extensions/pkgs/container/iscsi-tools/304494231) (ghcr.io/siderolabs/iscsi-tools@sha256:b57ff8145b0728cd60d1192bb2c48e43839044fe72e78a1cd6230b1839819d9e) released about a week after v0.1.6 and it appears to be working.

```
❯ bash create-rk1-installer-image.sh
Determining Talos version...
Talos client version v1.8.3 found. We will use that version for the Talos nodes, too.
Logging Docker into GitHub with user matthewstockdale...
Login Succeeded
Creating Talos image ghcr.io/matthewstockdale/talos-rk1-installer:v1.8.3 with provided extensions...
profile ready:
arch: arm64
platform: metal
secureboot: false
version: v1.8.3
input:
  kernel:
    path: /usr/install/arm64/vmlinuz
  initramfs:
    path: /usr/install/arm64/initramfs.xz
  baseInstaller:
    imageRef: ghcr.io/nberlee/installer:v1.8.3-rk3588
  overlayInstaller:
    imageRef: ghcr.io/nberlee/sbc-turingrk1:v1.8.3
  systemExtensions:
    - imageRef: ghcr.io/nberlee/rk3588:v1.8.3@sha256:61311adeff52826e659c1a80ad49940241bb08f3aaece80157834cf7f75cb789
    - imageRef: ghcr.io/siderolabs/iscsi-tools@sha256:b57ff8145b0728cd60d1192bb2c48e43839044fe72e78a1cd6230b1839819d9e
    - imageRef: ghcr.io/siderolabs/util-linux-tools:2.40.2@sha256:380740d159c024efa0f80136e7fdd18b36afdac77bad4f452aa83099ba115397
    - imageRef: ghcr.io/siderolabs/nut-client:2.8.2@sha256:547104892638321a88e5332c7b357a9a986b5f142afd247b201b4c726a97ac97
overlay:
  name: turingrk1
  image:
    imageRef: ghcr.io/nberlee/sbc-turingrk1:v1.8.3
output:
  kind: installer
  outFormat: raw
initramfs ready
kernel command line: talos.platform=metal console=tty0 console=ttyS9,115200 console=ttyS2,115200 cma=128MB sysctl.kernel.kexec_load_disabled=1 talos.dashboard.disabled=1 init_on_alloc=1 slab_nomerge pti=on consoleblank=0 nvme_core.io_timeout=4294967295 printk.devkmsg=on ima_template=ima-ng ima_appraise=fix ima_hash=sha512
installer container image ready
output asset path: /out/installer-arm64.tar
Pushing image ghcr.io/matthewstockdale/talos-rk1-installer:v1.8.3...
2024/11/19 20:38:15 existing blob: sha256:55270e74715a7c56d96b4384dc59d52048f14b192bc89b5f91ba5399212f2c07
2024/11/19 20:38:16 pushed blob: sha256:90022babc28e04cf17a3dc22353225b36bbed8834cd496d773043ff9db40d253
2024/11/19 20:38:29 pushed blob: sha256:13a309a1c8097609e9d9ab3b021343a39a7224cae07a59c709bcfa1673bb0098
2024/11/19 20:38:29 pushed blob: sha256:3a2ba28ffe28fb60398cec381c1936d1fa1c8e8cf00da806f4c64ac2270d0296
2024/11/19 20:38:31 pushed blob: sha256:30f918de180b26f9a9dec613bdb9862247bcca2a82f5cb9d00143900b4b41d59
2024/11/19 20:38:32 ghcr.io/matthewstockdale/talos-rk1-installer:v1.8.3: digest: sha256:d63fa1ef4d6d7170842c7026a2e60067b85a4f17b54eb212ff4017921d07560d size: 1086
ghcr.io/matthewstockdale/talos-rk1-installer@sha256:d63fa1ef4d6d7170842c7026a2e60067b85a4f17b54eb212ff4017921d07560d
```

I have successfully upgraded to 1.8.3 + iscsi-tools 0.1.6 and I am testing now.

**Comment by docbobo** [comment]
Thanks, @MatthewStockdale. I've been following the instructions provided by frezbo above, which gave me the following:

```
$ crane export ghcr.io/siderolabs/extensions:v1.8.3 | tar x -O image-digests | grep iscsi-tools
ghcr.io/siderolabs/iscsi-tools:v0.1.6@sha256:0c1ca0576ea6a5245b830b1d6093feafb751745a8efb0448bd170c068c88541f
```

This seems to be a different hash than the one you are quoting above.
